static/js/reactions.js

```diff
@@ -501,32 +521,115 @@ export function set_clean_reactions(message) {
     };
 
-    update_user_fields(r);
+    update_user_fields(r, message);
+    r.vote_text = get_vote_text_in_clean_reactions(r.user_ids, message);
 
     r.emoji_alt_code = user_settings.emojiset === "text";
     r.is_realm_emoji = r.reaction_type === "realm_emoji" || r.reaction_type === "zulip_extra_emoji";
 
     return r;
 }
 
-export function update_user_fields(r) {
+export function update_user_fields(r, message) {
     r.count = r.user_ids.length;
+    r.vote_text = get_vote_text(r.user_ids, message);
     r.label = generate_title(r.emoji_name, r.user_ids);
     if (r.user_ids.includes(page_params.user_id)) {
         r.class = "message_reaction reacted";
     } else {
         r.class = "message_reaction";
     }
 }
+
+// to set vote text when message is being cleaned.
+// this is an alternative to the commented out 'if'
+// block in get_total_votes_on_message() for now.
+function get_vote_text_in_clean_reactions(user_list, message) {
+    const helpers = get_helpers;
+    if (!helpers().should_display_emoji_reaction_users()) {
+        return `${user_list.length}`;
+    }
+    const usernames = people.get_display_full_names(user_list);
+    const current_user_has_reacted = user_list.includes(page_params.user_id);
+
+    if (current_user_has_reacted) {
+        const current_user_index = user_list.indexOf(page_params.user_id);
+        usernames[current_user_index] = $t({
+            defaultMessage: "You",
+        });
+    }
+
+    if (message.reactions.length <= 3) {
+        const comma_separated_usernames = usernames.join(", ");
+        return comma_separated_usernames;
+    }
+
+    // returning length as a string instead of integer for consistency
+    return `${usernames.length}`;
+}
+
+export function get_vote_text(user_list, message) {
+    const helpers = get_helpers;
+    if (!helpers().should_display_emoji_reaction_users()) {
+        return `${user_list.length}`;
+    }
+    const usernames = people.get_display_full_names(user_list);
+    const current_user_has_reacted = user_list.includes(page_params.user_id);
+
+    if (current_user_has_reacted) {
+        const current_user_index = user_list.indexOf(page_params.user_id);
+        usernames[current_user_index] = $t({
+            defaultMessage: "You",
+        });
+    }
+
+    if (get_total_votes_on_message(message) <= 3) {
+        const comma_separated_usernames = usernames.join(", ");
+        return comma_separated_usernames;
+    }
+
+    // returning length as a string instead of integer for consistency
+    return `${usernames.length}`;
+}
+
+export function get_total_votes_on_message(message) {
+    /*
+        this code path is followed when this function is called from
+        set_clean_reactions. At this point the message is still being
+        cleaned so we have to return the total votes in this manner.
+    */
+    // if (!message.message_reactions) {
+    //     return message.reactions.length;
+    // }
```

Comment:
Can you remove the commented out old versions here?  If you want to save an old implementation, use Git to store it, not comments in the version we're trying to merge.